new/release-process.md

```diff
@@ -0,0 +1,121 @@
+
+From Semantic Versioning:
+
+```
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when you make incompatible API changes,
+MINOR version when you add functionality in a backwards-compatible manner, and
+PATCH version when you make backwards-compatible bug fixes.
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+```
+
+Chef Client specific clarifications:
+
+```
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when the release will break existing functionality
+MINOR version when you add any new features
+PATCH version when you are only fixing bugs, typically regressions
+```
+
+##### Prerelease versions
+
+Any prerelease must be versioned in the format MAJOR.MINOR.PATCH.PRETYPE.PREVER, where:
+
+PRETYPE is the description of the prerelease that the Releaser chooses based on the state of the prerelease, e.g. alpha, beta, rc, dev.
+PREVER is an incremented number starting at zero.
+
+For example: 12.0.0.dev.0, 12.2.0.alpha.2, 12.3.0.rc.3
+
+#### Examples
+
+Major: When changing the load order of any cookbook segments (e.g. attributes, templates), the major version number shall be incremented.
+
+Minor: When adding support to the mount provider for special filesystem types that were previously unsupported, the minor version number shall be incremented.
+
+Patch: When a minor release is made that contains a new feature to a service provider that causes the service provider to no longer work for users on an older but still supported version of the platform, the patch version number shall be implemented.
+
+Changes that require the minor or patch version number to be implemented may be included in a release that increments a higher version number without incrementing a lower version number. That is, new features that don not exist in version 1.1.0 may be released in 2.0.0 without any intermediary releases.
+
+### Releasing
+
+#### Release Candidates
+
+Each major and minor release should have at least one week where a public RC is available
+for Chef users to test and report feedback.
+
+Patch releases are expected to have only small changes which are extensively tested, and thus may be safe to release without a release candidate.
+
+#### Chef Client Release Process
+
+* Verify the CHANGELOG.md is accurate for the current release
+* Update ```lib/chef/version.rb``` to the appropriate next version
+* Create an annotated git tag for the version
+* Trigger a build in CI (chef-trigger-release)
+* Write a blog post and share a public draft (e.g. a gist)
+
+Once the CI pipeline completes successfully:
+
+* Locally build and release the gems
+* Ask Chef Release Engineering to build and release AIX
+* Publish the blog post
+* Tweet the blog post
+* Email the chef and chef-dev mailing lists
```

Comment:
Not a blocker as this is accurate to the current process, but it would be nice to move the announce tasks a bit up the list as right now there is often a window where a new version is installable but not yet announced. The downside is that it would mean when the announce goes out, the new version might not be up for an hour or two, but maybe a pre-release email just to chef-dev? Something to make it slightly less confusing when Omnitruck suddenly starts serving a new version. Alternatively, kill omnitruck and add more explicit release control to packagecloud so this isn't needed :)